sdk/__init__.py

```diff
@@ -28,13 +28,20 @@ class SDK:
 
     def __init__(
         self,
-        username: str,
-        password: str,
         group_id: str,
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        login_url: Optional[str] = None,
```

Comment:
Should this have a convenient default value for the users ? (Is it even necessary anymore ?)

Reply:
To me I'd still prefer having the default being "" when possible

Reply:
Ha, I've changed everything now... ;) Why so?

Reply:
It's just an unnecessary type instability (in lack of a better word). For dict, str, list etc it's quite nice (and makes mypy extra happy) to leave them as their empty variants.

On the other hand, it's perhaps more user-instructing to mark them as Optional